python/src/spark_rapids_ml/utils.py

```diff
@@ -89,16 +89,32 @@ def _str_or_numerical(x: str) -> Union[str, float, int]:
 
 def _get_gpu_id(task_context: TaskContext) -> int:
     """Get the gpu id from the task resources"""
-    if task_context is None:
-        # safety check.
-        raise RuntimeError("_get_gpu_id should not be invoked from driver side.")
-    resources = task_context.resources()
-    if "gpu" not in resources:
-        raise RuntimeError(
-            "Couldn't get the gpu id, Please check the GPU resource configuration."
+    import os
+
+    if "CUDA_VISIBLE_DEVICES" in os.environ and os.environ["CUDA_VISIBLE_DEVICES"]:
+        num_assigned = len(os.environ["CUDA_VISIBLE_DEVICES"].split(","))
+        # when CUDA_VISIBLE_DEVICES is set and non-empty, use 0-th index entry
```

Comment:
I just tested, if I ran below commands in a bash

``` bash
export CUDA_VISIBLE_DEVICES=""

$ ./deviceQuery 
./deviceQuery Starting...

 CUDA Device Query (Runtime API) version (CUDART static linking)

cudaGetDeviceCount returned 100
-> no CUDA-capable device is detected
Result = FAIL
```

I guess, It will throw an exception if we use gpu_id=0.

Reply:
Good point.   Ideally we don't get this situation, but we should raise an exception right away in this case instead of simply assigning based on the TaskContext.